pkg/agent/client.go

```diff
@@ -479,25 +486,35 @@ func (a *Client) remoteToProxy(connID int64, ctx *connContext) {
 		klog.V(5).InfoS("received data from remote", "bytes", n, "connectionID", connID)
 
 		if err == io.EOF {
-			klog.V(2).Infoln("connection EOF")
+			klog.V(2).InfoS("connection EOF", "connectionID", connID)
 			return
 		} else if err != nil {
 			// Normal when receive a CLOSE_REQ
-			klog.ErrorS(err, "connection read failure")
+			klog.ErrorS(err, "connection read failure", "connectionID", connID)
 			return
 		} else {
 			resp.Payload = &client.Packet_Data{Data: &client.Data{
 				Data:      buf[:n],
 				ConnectID: connID,
 			}}
 			if err := a.Send(resp); err != nil {
-				klog.ErrorS(err, "stream send failure")
+				klog.ErrorS(err, "stream send failure", "connectionID", connID)
 			}
 		}
 	}
 }
 
 func (a *Client) proxyToRemote(connID int64, ctx *connContext) {
+	defer func() {
+		if err := recover(); err != nil {
```

Comment:
Not sure `error` is the correct name for the return value of `recover()` because it's more like info from panic, instead of an error caused by `recover()`

Reply:
I agree "error" is close but not quite right. "Info" is to vague. "panic"? "exception"?

Reply:
panicInfo to be specific?

Reply:
In Golang's [blog](https://go.dev/blog/defer-panic-and-recover) they just use 'r'. I think `err` is ok. After all, this is just a local variable.

Reply:
I think we are also implicitly discussing the structured logging name. So in that sense it is more than just the local variable.